pcdsutils/qt/lazy.py

```diff
@@ -0,0 +1,65 @@
+from typing import ClassVar, Optional, Type
+
+from qtpy import QtCore, QtGui, QtWidgets
+
+
+class LazyWidget(QtWidgets.QWidget):
+    """
+    A lazy widget which only is created when first shown.
+
+    Parameters
+    ----------
+    widget_cls : QtWidgets.QWidget subclass
+        The widget class to instantiate.
+    """
+
+    widget_cls: Type[QtWidgets.QWidget]
+    widget: Optional[QtWidgets.QWidget]
+
+    widget_created: ClassVar[QtCore.Signal] = QtCore.Signal(QtWidgets.QWidget)
+    widget_shown: ClassVar[QtCore.Signal] = QtCore.Signal()
+    widget_hidden: ClassVar[QtCore.Signal] = QtCore.Signal()
+
+    def __init__(self, widget_cls: Type[QtWidgets.QWidget]):
+        super().__init__()
+        self.widget_cls = widget_cls
+        self.widget = None
+
+        self.setVisible(False)
+        self.setLayout(QtWidgets.QVBoxLayout())
```

Comment:
I see no reason not to merge this utility, but one small thought: does this create any subtle layout problems with the default margins of `QVBoxLayout`?

Reply:
I think there are improvements that could be made with respect to layout padding/margins, for sure. The subtleties were not investigated at all.

I'd be OK with either:
* Iterating on this prior to merging (as we have no immediate need)
* Or merging and then circling back (when the need arises)